### ISSUE TITLE
Error when deploying to OpenShift

### ISSUE THREAD

**Issue by gsmet** [quarkusio/quarkus#41625]
I'm not sure if it's due to an update to the cluster/an update in Quarkus or whatever but I didn't have this error until now:

That didn't prevent the app from being deployed.

/cc @iocanel @manusa @metacosm 

```
[ERROR] listSyncAndWatch failed for v1/namespaces/prod-quarkus-bot/services, will stop
java.util.concurrent.CompletionException: java.util.concurrent.RejectedExecutionException
    at java.util.concurrent.CompletableFuture.encodeThrowable (CompletableFuture.java:315)
    at java.util.concurrent.CompletableFuture.completeThrowable (CompletableFuture.java:320)
    at java.util.concurrent.CompletableFuture$UniCompose.tryFire (CompletableFuture.java:1159)
    at java.util.concurrent.CompletableFuture.postComplete (CompletableFuture.java:510)
    at java.util.concurrent.CompletableFuture.complete (CompletableFuture.java:2147)
    at io.fabric8.kubernetes.client.http.StandardHttpClient.lambda$completeOrCancel$10 (StandardHttpClient.java:142)
    at java.util.concurrent.CompletableFuture.uniWhenComplete (CompletableFuture.java:863)
    at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire (CompletableFuture.java:841)
    at java.util.concurrent.CompletableFuture.postComplete (CompletableFuture.java:510)
    at java.util.concurrent.CompletableFuture.complete (CompletableFuture.java:2147)
    at io.fabric8.kubernetes.client.http.ByteArrayBodyHandler.onBodyDone (ByteArrayBodyHandler.java:51)
    at java.util.concurrent.CompletableFuture.uniWhenComplete (CompletableFuture.java:863)
    at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire (CompletableFuture.java:841)
    at java.util.concurrent.CompletableFuture.postComplete (CompletableFuture.java:510)
    at java.util.concurrent.CompletableFuture.complete (CompletableFuture.java:2147)
    at io.fabric8.kubernetes.client.vertx.VertxHttpRequest.lambda$null$1 (VertxHttpRequest.java:121)
    at io.vertx.core.impl.ContextInternal.dispatch (ContextInternal.java:279)
    at io.vertx.core.impl.ContextInternal.dispatch (ContextInternal.java:261)
    at io.vertx.core.http.impl.HttpEventHandler.handleEnd (HttpEventHandler.java:76)
    at io.vertx.core.http.impl.HttpClientResponseImpl.handleEnd (HttpClientResponseImpl.java:250)
    at io.vertx.core.http.impl.Http1xClientConnection$StreamImpl.lambda$new$0 (Http1xClientConnection.java:421)
    at io.vertx.core.streams.impl.InboundBuffer.handleEvent (InboundBuffer.java:255)
    at io.vertx.core.streams.impl.InboundBuffer.write (InboundBuffer.java:134)
    at io.vertx.core.http.impl.Http1xClientConnection$StreamImpl.handleEnd (Http1xClientConnection.java:709)
    at io.vertx.core.impl.ContextImpl.execute (ContextImpl.java:313)
    at io.vertx.core.impl.ContextImpl.execute (ContextImpl.java:293)
    at io.vertx.core.http.impl.Http1xClientConnection.handleResponseEnd (Http1xClientConnection.java:940)
    at io.vertx.core.http.impl.Http1xClientConnection.handleHttpMessage (Http1xClientConnection.java:810)
    at io.vertx.core.http.impl.Http1xClientConnection.handleMessage (Http1xClientConnection.java:774)
    at io.vertx.core.net.impl.ConnectionBase.read (ConnectionBase.java:159)
    at io.vertx.core.net.impl.VertxHandler.channelRead (VertxHandler.java:153)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead (AbstractChannelHandlerContext.java:442)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead (AbstractChannelHandlerContext.java:420)
    at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead (AbstractChannelHandlerContext.java:412)
    at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead (CombinedChannelDuplexHandler.java:436)
    at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead (ByteToMessageDecoder.java:346)
    at io.netty.handler.codec.ByteToMessageDecoder.channelRead (ByteToMessageDecoder.java:318)
    at io.netty.channel.CombinedChannelDuplexHandler.channelRead (CombinedChannelDuplexHandler.java:251)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead (AbstractChannelHandlerContext.java:442)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead (AbstractChannelHandlerContext.java:420)
    at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead (AbstractChannelHandlerContext.java:412)
    at io.netty.handler.ssl.SslHandler.unwrap (SslHandler.java:1475)
    at io.netty.handler.ssl.SslHandler.decodeJdkCompatible (SslHandler.java:1338)
    at io.netty.handler.ssl.SslHandler.decode (SslHandler.java:1387)
    at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection (ByteToMessageDecoder.java:530)
    at io.netty.handler.codec.ByteToMessageDecoder.callDecode (ByteToMessageDecoder.java:469)
    at io.netty.handler.codec.ByteToMessageDecoder.channelRead (ByteToMessageDecoder.java:290)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead (AbstractChannelHandlerContext.java:444)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead (AbstractChannelHandlerContext.java:420)
    at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead (AbstractChannelHandlerContext.java:412)
    at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead (DefaultChannelPipeline.java:1410)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead (AbstractChannelHandlerContext.java:440)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead (AbstractChannelHandlerContext.java:420)
    at io.netty.channel.DefaultChannelPipeline.fireChannelRead (DefaultChannelPipeline.java:919)
    at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read (AbstractNioByteChannel.java:166)
    at io.netty.channel.nio.NioEventLoop.processSelectedKey (NioEventLoop.java:788)
    at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized (NioEventLoop.java:724)
    at io.netty.channel.nio.NioEventLoop.processSelectedKeys (NioEventLoop.java:650)
    at io.netty.channel.nio.NioEventLoop.run (NioEventLoop.java:562)
    at io.netty.util.concurrent.SingleThreadEventExecutor$4.run (SingleThreadEventExecutor.java:997)
    at io.netty.util.internal.ThreadExecutorMap$2.run (ThreadExecutorMap.java:74)
    at io.netty.util.concurrent.FastThreadLocalRunnable.run (FastThreadLocalRunnable.java:30)
    at java.lang.Thread.run (Thread.java:840)
Caused by: java.util.concurrent.RejectedExecutionException
    at io.fabric8.kubernetes.client.utils.internal.SerialExecutor.execute (SerialExecutor.java:47)
    at io.fabric8.kubernetes.client.informers.impl.cache.SharedProcessor.execute (SharedProcessor.java:179)
    at io.fabric8.kubernetes.client.informers.impl.cache.Reflector.lambda$null$4 (Reflector.java:136)
    at io.fabric8.kubernetes.client.informers.impl.cache.ProcessorStore.retainAll (ProcessorStore.java:116)
    at io.fabric8.kubernetes.client.informers.impl.cache.Reflector.lambda$listSyncAndWatch$6 (Reflector.java:130)
    at java.util.concurrent.CompletableFuture$UniCompose.tryFire (CompletableFuture.java:1150)
    at java.util.concurrent.CompletableFuture.postComplete (CompletableFuture.java:510)
    at java.util.concurrent.CompletableFuture.complete (CompletableFuture.java:2147)
    at io.fabric8.kubernetes.client.http.StandardHttpClient.lambda$completeOrCancel$10 (StandardHttpClient.java:142)
    at java.util.concurrent.CompletableFuture.uniWhenComplete (CompletableFuture.java:863)
    at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire (CompletableFuture.java:841)
    at java.util.concurrent.CompletableFuture.postComplete (CompletableFuture.java:510)
    at java.util.concurrent.CompletableFuture.complete (CompletableFuture.java:2147)
    at io.fabric8.kubernetes.client.http.ByteArrayBodyHandler.onBodyDone (ByteArrayBodyHandler.java:51)
    at java.util.concurrent.CompletableFuture.uniWhenComplete (CompletableFuture.java:863)
    at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire (CompletableFuture.java:841)
    at java.util.concurrent.CompletableFuture.postComplete (CompletableFuture.java:510)
    at java.util.concurrent.CompletableFuture.complete (CompletableFuture.java:2147)
    at io.fabric8.kubernetes.client.vertx.VertxHttpRequest.lambda$null$1 (VertxHttpRequest.java:121)
    at io.vertx.core.impl.ContextInternal.dispatch (ContextInternal.java:279)
    at io.vertx.core.impl.ContextInternal.dispatch (ContextInternal.java:261)
    at io.vertx.core.http.impl.HttpEventHandler.handleEnd (HttpEventHandler.java:76)
    at io.vertx.core.http.impl.HttpClientResponseImpl.handleEnd (HttpClientResponseImpl.java:250)
    at io.vertx.core.http.impl.Http1xClientConnection$StreamImpl.lambda$new$0 (Http1xClientConnection.java:421)
    at io.vertx.core.streams.impl.InboundBuffer.handleEvent (InboundBuffer.java:255)
    at io.vertx.core.streams.impl.InboundBuffer.write (InboundBuffer.java:134)
    at io.vertx.core.http.impl.Http1xClientConnection$StreamImpl.handleEnd (Http1xClientConnection.java:709)
    at io.vertx.core.impl.ContextImpl.execute (ContextImpl.java:313)
    at io.vertx.core.impl.ContextImpl.execute (ContextImpl.java:293)
    at io.vertx.core.http.impl.Http1xClientConnection.handleResponseEnd (Http1xClientConnection.java:940)
    at io.vertx.core.http.impl.Http1xClientConnection.handleHttpMessage (Http1xClientConnection.java:810)
    at io.vertx.core.http.impl.Http1xClientConnection.handleMessage (Http1xClientConnection.java:774)
    at io.vertx.core.net.impl.ConnectionBase.read (ConnectionBase.java:159)
    at io.vertx.core.net.impl.VertxHandler.channelRead (VertxHandler.java:153)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead (AbstractChannelHandlerContext.java:442)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead (AbstractChannelHandlerContext.java:420)
    at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead (AbstractChannelHandlerContext.java:412)
    at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead (CombinedChannelDuplexHandler.java:436)
    at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead (ByteToMessageDecoder.java:346)
    at io.netty.handler.codec.ByteToMessageDecoder.channelRead (ByteToMessageDecoder.java:318)
    at io.netty.channel.CombinedChannelDuplexHandler.channelRead (CombinedChannelDuplexHandler.java:251)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead (AbstractChannelHandlerContext.java:442)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead (AbstractChannelHandlerContext.java:420)
    at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead (AbstractChannelHandlerContext.java:412)
    at io.netty.handler.ssl.SslHandler.unwrap (SslHandler.java:1475)
    at io.netty.handler.ssl.SslHandler.decodeJdkCompatible (SslHandler.java:1338)
    at io.netty.handler.ssl.SslHandler.decode (SslHandler.java:1387)
    at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection (ByteToMessageDecoder.java:530)
    at io.netty.handler.codec.ByteToMessageDecoder.callDecode (ByteToMessageDecoder.java:469)
    at io.netty.handler.codec.ByteToMessageDecoder.channelRead (ByteToMessageDecoder.java:290)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead (AbstractChannelHandlerContext.java:444)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead (AbstractChannelHandlerContext.java:420)
    at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead (AbstractChannelHandlerContext.java:412)
    at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead (DefaultChannelPipeline.java:1410)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead (AbstractChannelHandlerContext.java:440)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead (AbstractChannelHandlerContext.java:420)
    at io.netty.channel.DefaultChannelPipeline.fireChannelRead (DefaultChannelPipeline.java:919)
    at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read (AbstractNioByteChannel.java:166)
    at io.netty.channel.nio.NioEventLoop.processSelectedKey (NioEventLoop.java:788)
    at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized (NioEventLoop.java:724)
    at io.netty.channel.nio.NioEventLoop.processSelectedKeys (NioEventLoop.java:650)
    at io.netty.channel.nio.NioEventLoop.run (NioEventLoop.java:562)
    at io.netty.util.concurrent.SingleThreadEventExecutor$4.run (SingleThreadEventExecutor.java:997)
    at io.netty.util.internal.ThreadExecutorMap$2.run (ThreadExecutorMap.java:74)
    at io.netty.util.concurrent.FastThreadLocalRunnable.run (FastThreadLocalRunnable.java:30)
    at java.lang.Thread.run (Thread.java:840)

```

**Comment by geoand** [comment]
> but I didn't have this error until now

A missed opportunity to use cool words such as `hitherto` and `heretofore` :)

**Comment by manusa** [comment]
Seems to be the same as #6059 which has already been fixed and released in 6.13.1.

#41605 should take care of this, once merged, this issue shouldn't be reproducible anymore/hencefore/hereafter.

> A missed opportunity to use cool words such as hitherto and heretofore :)

Never thought Quarkus' GitHub issues would be the place for me to improve my English :top:

**Comment by geoand** [comment]
> Never thought Quarkus' GitHub issues would be the place for me to improve my English 🔝

😆
